apiserver/logsink/logsink.go

```diff
@@ -156,6 +167,9 @@ func (h *logSinkHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				}
 			case m, ok := <-logCh:
 				if !ok {
+					h.mu.Lock()
+					defer h.mu.Unlock()
```

Comment:
I think this is clearer as
mu.Lock()
var = true
mu.Unlock()

no need for defer when you are done right away.